src/extensions/String.test.ts

```diff
@@ -0,0 +1,166 @@
+import { describe } from 'riteway'
+
+import './String'
+
+describe('String.prototype.firstAndLastCharacters', async ( should: any) => {
+  const { assert } = should()
+
+  {
+    const testString = 'testmeonlongstring'
+    assert({
+      given: 'number 6 and string longer than length 12',
+      should: 'return the 6 fist and last characters of the string seperated by ...',
+      actual: testString.firstAndLastCharacters(6),
+      expected: 'testme...string',
+    })
+  }
+
+  {
+    const testString = 'teststring'
+    assert({
+      given: 'number 6 and string shorter than length 12',
+      should: 'return the 6 fist and last characters of the string seperated by ...',
+      actual: testString.firstAndLastCharacters(6),
+      expected: 'testst...string',
+    })
+  }
+
+  {
+    const testString = 'test'
+    assert({
+      given: 'number 6 and string shorter than length 6',
+      should: 'return the string 2 times seperated by ...',
+      actual: testString.firstAndLastCharacters(6),
+      expected: 'test...test',
+    })
+  }
+})
+
+describe('String.prototype.trimLeftByPattern', async (should: any) => {
+  const { assert } = should()
+
+  {
+    const testString = 'bad/init'
+    
+    assert({
+      given: 'string that does not begin with supplied pattern',
+      should: 'return string',
+      actual: testString.trimLeftByPattern('/'),
+      expected: 'bad/init',
+    })
+  }
+
+  {
+    const testString = '/good/init'
+    
+    assert({
+      given: 'string that begins with supplied pattern',
+      should: 'return string',
+      actual: testString.trimLeftByPattern('/'),
+      expected: 'good/init',
+    })
+  }
+})
+
+describe('String.prototype.padEnd', async (should: any) => {
+  const { assert } = should()
+
+  {
+    const testString = 't'
+    
+    assert({
+      given: 'string shorter than supplied targetLength + long padString',
+      should: 'return correct string',
+      actual: testString.padEnd(5, 'thisistoolong'),
+      expected: 'tthis',
+    })
+  }
+
+  {
+    const testString = 't'
+    
+    assert({
+      given: 'string shorter than supplied targetLength + padString',
+      should: 'return correct string',
+      actual: testString.padEnd(5, 'this'),
+      expected: 'tthis',
+    })
+  }
+
+  {
+    const testString = 't'
+    
+    assert({
+      given: 'string shorter than supplied targetLength + short padString',
+      should: 'return correct string',
```

Comment:
maybe something like "should: 'return a string with repeated characters from the pad string'".
That would give a good description of the behavior when the pad string is short.